src/nvim/api/buffer.c

```diff
@@ -179,6 +179,11 @@ void buffer_set_line_slice(Buffer buffer,
     return;
   }
 
+  if (!inbounds(buf, start)) {
+    api_set_error(err, Validation, _("Index out of bounds"));
+    return;
```

Comment:
Before this change `buffer_inser(buffer,-n-1,lines,err)` could be used to insert lines at the beginning of the buffer (n = number of lines in the buffer). After the change this is only possible with
`buffer_set_line_slice(buffer,0,0,true,false,replacement,err);`.